libethash-cl/CLMiner.cpp

```diff
@@ -672,15 +679,60 @@ bool CLMiner::init(int epoch)
             return false;
         }
 
+        /* If we have a binary kernel, we load it in tandem with the opencl,
+		   that way, we can use the dag generate opencl code and fall back on
+           the default kernel if loading fails for whatever reason */
+		bool loadedBinary = false;
+
+        if(s_clKernelName == CLKernelName::Binary) {
+            std::ifstream kernel_file;
+            vector<unsigned char> bin_data;
+            std::stringstream fname_strm;
+
+            /* Open kernels/ethash_{devicename}_lws{local_work_size}.bin */
+            std::transform(device_name.begin(), device_name.end(), device_name.begin(), ::tolower);
+            fname_strm << "kernels/ethash_" << device_name << "_lws" << m_workgroupSize << ".bin";
```

Comment:
Kernels should live in a place relative to install forlder, ie:

fname_strm << boost::dll::program_location().parent_path().string() << "/kernels/" << name << s_workGroupSize << ".bin";